builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -294,6 +304,38 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 				return err
 			}
 		}
+
+		if clusterUpdate {
+			log.Printf("[INFO] RDS Cluster is restoring from snapshot with default db_cluster_parameter_group_name and backup_retention_period, " +
+				"but custom values should be set, will now update after snapshot is restored!")
+
+			d.SetId(d.Get("cluster_identifier").(string))
+
+			log.Printf("[INFO] RDS Cluster ID: %s", d.Id())
+
+			log.Println("[INFO] Waiting for RDS Cluster to be available")
+
+			stateConf := &resource.StateChangeConf{
+				Pending:    []string{"creating", "backing-up", "modifying"},
+				Target:     []string{"available"},
+				Refresh:    resourceAwsRDSClusterStateRefreshFunc(d, meta),
+				Timeout:    120 * time.Minute,
+				MinTimeout: 3 * time.Second,
+				Delay:      30 * time.Second, // Wait 30 secs before starting
+			}
+
+			// Wait, catching any errors
+			_, err = stateConf.WaitForState()
+			if err != nil {
+				return err
+			}
+
+			err = resourceAwsRDSClusterUpdate(d, meta)
+			if err != nil {
+				return err
+			}
```

Comment:
Hi! That's a possibility, it just doesn't seem to fit in nicely with the rest of the coding style, that's why I used the same logic as in the sgUpdate block.

If we change the logic for clusterUpdate we should also change it for sgUpdate for consistency. But since that would involve making more changes than the minimum necessary to fix the bug I'm not sure if that's the correct course of action. 

Also, both clusterUpdate and sgUpdate only make sense if restoring from snapshot (first if statement). When creating as a replica or with a username/password combination (other 2 else statements) they don't apply. The overall approach on this function seems to be to isolate variables and functionality inside the if statements and then terminate the function with a final refresh to ensure the resource is available followed by a read to return the current cluster settings.

It is probably a bit of extra code but it adds to clarity. I'm happy to make the changes though if that's what you prefer.